[사다리게임] 스텝 2 완료

hngfu opened the pull request:
메소드 하나가 하나의 기능을 하도록 만들고 약간 길어진 메소드는 분리하였습니다.

hngfu committed:
사다리 출력 부분 더 자세히 분리

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -61,14 +61,18 @@ func removeConnectedGripOf(_ ladder:[String]) -> [String]{
+
 func main(){
     print("참여할 사람은 몇 명 인가요?(2이상의 수를 입력해주세요.)")
     guard let participant = readLine() else { return }
```
참여할 사람을 입력 받는 기능도 함수로 분리해보세요.
이번 단계에서 더 작은 단위로 함수를 분리하는 연습을 해야합니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -80,9 +84,9 @@ func main(){
 
```
convert()는 무엇을 무엇으로 바꾸는건가요? 
이게 바로 위에 입력과 함께 처리하는게 좋은지 확인해보세요.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -28,7 +28,7 @@ func makeLadder(_ participant:Int, _ maximumLadderHeight:Int) -> [[String]] {
-
+    
     return ladder
 }
```
makeLadder() 함수가 [[String]] 리턴값을 만들어냅니다. 
사다리 정보를 표현하는데 String 타입이 충분한가요? 
데이터를 표현하기 위한 타입을 결정하는 것은 매우 중요합니다. 
특히 사다리 정보를 표현하기 위해 충분한 데이터 타입을 선택하는 기준을 정해보세요.
단지 출력하는 것을 위해서 데이터 타입을 선택하지 마세요.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -61,14 +61,18 @@ func removeConnectedGripOf(_ ladder:[String]) -> [String]{
-            print(j, terminator: "")
-        }
+        showLayerOf(ladder: i)
         print("|")
```
showLayerOf()가 한 줄을 표현하는 거라면 여기 `print("|")` 도 showLayerOf() 내부에 포함되어야 의미있지 않을까요?

godrm reviewed (commented):
이번 단계는 최대한 함수를 분리해보는 연습이 필요합니다.
다음 피드백을 확인해보세요.

hngfu commented:
피드백 감사합니다!
수정 후 다시 말씀드리겠습니다!!

hngfu committed:
[사다리게임] String 타입 대신 Bool 타입 이용

사다리 만드는데 사용하는 타입을 Bool으로 변경했습니다.
함수를 한가지 일만 하도록 만들었습니다.
뎁스를 전부 1이 넘지 않도록 하였습니다.

hngfu commented:
사다리 게임 2단계 수정 완료하였습니다.
전에 피드백으로 커밋을 자주 해서 로그만 보고도 무엇을 변경하였는지 알도록 하라고 말씀하셨는데
어쩌다 보니 한번에 하게 되었습니다.
다음에는 커밋을 더 자주 하도록 하겠습니다!

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -8,86 +8,87 @@
+func removeConnectedGripOf(_ ladder:[Bool]) -> [Bool] {
     var ladderWithRemovedGrip = ladder
-    
     for i in 0..<ladderWithRemovedGrip.count - 1 {
```
for 반복문을 만들때 다양한 방법을 고민해보세요. count 를 쓰기 보다는 index를 쓰는 방법을 권장합니다. 
성능상에도 차이가 있으니까 이 부분을 확인해보세요.

hngfu committed:
[사다리게임] for문 사용시 'count'대신 'index'을 사용하도록 변경

전에는 for문을 사용할 때 0부터 해당배열의 갯수로 범위를 지정했지만
지금은 startIndex와 endIndex를 이용하여 for문이 돌아가게 변경하였습니다.

hngfu commented:
'count'대신 'startIndex'와 'endIndex'로 바꾸었습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -8,86 +8,87 @@
+    let ladderHeight = convertStringToInt(inputValue())
+    
+    if checkMinimum(number: participant) && checkMinimum(number: ladderHeight) {
+        show(ladder: makeLadder(participant, ladderHeight))
```
`show(ladder: makeLadder(participant, ladderHeight))` 처럼 함수를 중첩해서 쓰기보다는
이번 미션까지는 한 줄에 하나의 함수를 호출하도록 해보세요.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -8,86 +8,87 @@
-
-    if checkMinimum(number: convertedParticipant) && checkMinimum(number: convertedLadderHeight) {
-        show(ladder: makeLadder(convertedParticipant, convertedLadderHeight))
+    let ladderHeight = convertStringToInt(inputValue())
```
입력 받는것과 참여자수, 사다리수로 바꾸는 것을 합쳐보면 어떨까요?
예를 들어 참여자 수를 입력받아 숫자로 리턴해주는거죠. 지금과 어떤게 좋은 선택일까요?

hngfu committed:
[사다리게임] 함수 매개변수로 함수를 바로 사용하지 않게 수정

함수를 바로 중첩해서 사용하지 않게 수정하였습니다.

hngfu committed:
[사다리게임] 입력받는 함수가 바로 Int형으로 변환시켜서 리턴하도록 수정

입력받는 함수에서 리턴할때 Int형으로 바뀌고 리턴하도록 수정하였습니다.

hngfu committed:
[사다리게임] return이 빠져있어서 붙여줌

삭제되어있던 return을 다시 붙여주었습니다.

hngfu commented:
피드백 주신대로 입력받고 바로 형변환시켜주고 리턴하는게 가독성과 여러 방면으로 훨씬 더 좋은것 같습니다.
수정완료하였습니다.

godrm commented:
커밋할때 이미 저장소가 사다리게임이니 `[]` 말머리는 다른 정보를 표시하는게 좋겠습니다.

godrm commented on LadderGame/LadderGame/main.swift:
```diff
@@ -87,6 +87,7 @@ func main(){
     if checkMinimum(number: participant) && checkMinimum(number: ladderHeight) {
         let ladder = makeLadder(participant, ladderHeight)
         show(ladder)
+        return
```
그래서 이렇게 조건이 맞을때 if 문 아래 범위에서 실행하는_ 로직보다는
guard else로 조건이 맞으면 아래에서 실행하도록 하고, 조건이 맞지 않으면 먼저 빠져나가도록 하면 오히려 실수를 줄일 수 있습니다. 참고하세요.

hngfu replied:
알려주셔서 감사합니다!
